internal/controller/storageclient_controller.go

```diff
@@ -411,6 +410,15 @@ func (r *storageClientReconcile) reconcilePhases() (ctrl.Result, error) {
 		return reconcile.Result{}, err
 	}
 
+	if controllerutil.AddFinalizer(&r.storageClient, storageClientFinalizer) {
+		r.storageClient.Status.Phase = v1alpha1.StorageClientInitializing
```

Comment:
Why are we setting this here? shouldn't we be in initializing phase unrelated to the finalizer?